pkg/rscache/file.go

```diff
@@ -185,12 +194,18 @@ func (o *fileCache) Head(ctx context.Context, resolver ResolverSpec) (size int64
 
 		// Wait
 		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
```

Comment:
do we need `default` here?

Reply:
@nodivbyzero Let me experiment with it some more, but if I'm remembering correctly, I was experiencing some test failures if I left out the `default` statement.

Reply:
@nodivbyzero it looks like removing the `default` statement causes the following code to be unreachable and results in a deadlock.